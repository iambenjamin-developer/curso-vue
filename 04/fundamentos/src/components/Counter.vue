<template>

    <h2>{{ customTitleV3 }}</h2>
    <p> {{ counter }} <sup>2</sup> = {{ counter * counter }}</p>

    <div>
        <button v-on:click="increase">+1</button>
        <button @click="decrease">-1</button>
    </div>

</template>

<script>
export default {
    props:
    {
        title: String,
        start: {
            type: Number,
            default: 7,
            // required: true
            validator(value) {
                return value > -1;
            }
        }
    },
    data() {
        return {
            counter: this.start
        }
    },
    methods: {
        getSquareValue() {
            console.log("method => getSquareValue")
            return this.counter * this.counter
        },
        increase() {
            // this.counter++;
            this.counter = this.counter + 1;
        },
        decrease() {
            this.counter--;
        }
    },
    computed: {
        getSquareCounterValue() {
            console.log("computed => getSquareCounterValue")
            return this.counter * this.counter
        },
        customTitle() {
            return this.title || 'Titulo por defecto'
        },
        customTitleV2() {
            if (this.title !== undefined) {
                return this.title;
            } else {
                return 'Título por defecto';
            }
        },
        customTitleV3() {
            return this.title ? this.title : 'Título por defecto';
        },
    }
}
</script>

<style>
button {
    background-color: #64BB87;
    border-radius: 5px;
    border: 1px solid white;
    color: white;
    cursor: pointer;
    margin: 0 5px;
    padding: 5px 15px;
    transition: 0.3s ease-in-out;
}

button:hover {
    background-color: #003b19;
    transition: 0.3s ease-in-out;
}
</style>