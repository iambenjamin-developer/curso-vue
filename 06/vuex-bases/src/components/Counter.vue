<template>
    <h1>Counter Vuex</h1>
    <h2>Acceso directo {{ $store.state.count }}</h2>

    <button @click="incrementarEnUno">+1</button>
    <button @click="incrementarEnCinco">+5</button>
    <input type="text" @keydown.enter="incrementarPorValorParametro(valor)" v-model="valor" />
</template>

<script>

export default {

    data() {
        return {

            valor: 0
        }
    },
    methods:
    {
        incrementarEnUno() {

            this.$store.commit('incrementOne')
        },
        incrementarEnCinco() {

            this.$store.commit('incrementFive')
        },
        incrementarPorValorParametro(valor) {


            this.$store.commit('incrementByValue', valor)
        },
    }

}
</script>
