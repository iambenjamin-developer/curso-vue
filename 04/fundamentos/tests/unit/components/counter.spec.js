import { shallowMount } from '@vue/test-utils'
import Counter from '@/components/Counter'

describe('Counter Component', () => {

    // test('debe hacer match con el snapshot', () => {

    //     const wrapper = shallowMount(Counter);

    //     expect(wrapper.html()).toMatchSnapshot();
    // })

    // test('h2 debe de tener el valor por defecto "Titulo por defecto!"', () => {

    //     const wrapper = shallowMount(Counter);

    //     expect(wrapper.find('h2').exists()).toBeTruthy();

    //     // expect(wrapper.find('h20').exists()).toBe(true);

    //     const h2Value = wrapper.find('h2').text();

    //     expect(h2Value).toBe("Titulo por defecto!")

    // })

    test('El valor por defecto debe de ser 100 en el <p> (con findAll)', () => {

        //Wrapper
        const wrapper = shallowMount(Counter);

        //pTags
        const pTags = wrapper.findAll('p');

        // console.log(pTags[0].text())
        //expect segundo p = 100
        expect(pTags[1].text()).toBe('100');

    })

    test('El valor por defecto debe de ser 100 en el <p> (con data attribute)', () => {

        const wrapper = shallowMount(Counter);

        const value = wrapper.find('[data-testid="counter"]').text();

        expect(value).toBe('100');

    })

    test('Debe de incrementar en 1 el valor del contador', async () => {

        const wrapper = shallowMount(Counter);

        const increaseBtn = wrapper.find('button');

        await increaseBtn.trigger('click');

        const value = wrapper.find('[data-testid="counter"]').text();

        expect(value).toBe('101');

    })
})