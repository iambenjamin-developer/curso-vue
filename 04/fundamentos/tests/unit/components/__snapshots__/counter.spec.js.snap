// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component debe hacer match con el snapshot 1`] = `
<h2>Titulo por defecto!</h2>
<p>7 <sup>2</sup> = function () { [native code] }</p>
<div><button>+1</button><button>-1</button></div>
`;
